_source/_posts/2021-12-03-angular-dynamic-components.md

```diff
@@ -0,0 +1,773 @@
+
+In [Angular](https://angular.io), you can show different components or even parts of templates conditionally using built-in directives such as `*ngIf`. But as the number of conditions to evaluate or the differences in the view changes, managing the correct view to display becomes difficult. These types of scenarios are where [dynamic components](https://angular.io/guide/dynamic-component-loader) are helpful. Angular has the mechanics to load components at runtime so you can dynamically display content. 
+
+ After following the instructions in the post, you'll have an Angular app using Material controls that displays unique content based on an authenticated user's claim value.
+
+The main flow for the application is after initially launching a welcome page; you'll log in using Okta from a button in the toolbar and redirect to the guarded content. The toolbar now displays your name, and you'll see a rotation of dynamically created components based on a user claim from your ID token.
+
+{% img blog/angular-dynamic-components/overview.gif alt:"animated image showing logging in and displaying dynamic contents" width:"800" %}{: .center-image }
+
+In this post, we'll
+* Set up an Angular application with routing and lazy-loaded modules
+* Create an OIDC client in Okta
+* Add authentication to the Angular application
+* Capture authenticated user information and user claims
+* Simulate an external server call and implement the dynamic components
+* Complete displaying the dynamic components using the user claim value
+
+We'll be covering a lot in this post to build out a non-trivial Angular application. As a result, this post assumes some Angular knowledge. We'll be blazing through some parts to spend more time working through the dynamic components and user profile information.
+
+If you are new to Angular, check out the following fantastic resources and guides first
+* [Angular Quickstart](https://angular.io/guide/what-is-angular)
+* [A Quick Guide to Angular and GraphQL](/blog/2021/10/22/angular-graphql#integrate-oidc-for-auth)
+
+**Prerequisites**
+* A valid version of [Node](https://nodejs.org/) for Angular v13, such as Node 14.15
+* [Angular CLI](https://cli.angular.io)
+* [Okta CLI](https://cli.okta.com/)
+
+{% include toc.md %}
+
+## Scaffold the Angular application
+
+The [Angular CLI](https://cli.angular.io) automates creating Angular apps quickly. It makes the project folder and scaffolds the application shell. We'll pass in the parameters to
+* add routing
+* use scss for styles
+* use inline templates
+* use inline styles
+* skip tests (the code repo includes tests to show working tests with testbed setup and authentication service spies, so feel free to include tests if you want to try that yourself)
```

Comment:
```suggestion
* skip tests (The code repo includes tests to show working tests with testbed setup and authentication service spies, so feel free to include tests if you want to try that yourself.)
```
The sentence in parens needs to be treated as a full sentence (which it is). Or, you could add it after the bulleted list as a "Note" or parenthetical standalone sentence.